src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx

```diff
@@ -0,0 +1,575 @@
+      }
+      batchFileRequest.requestedFiles.push(requestObj)
+      if (!requestedFileTypes[dataAccessRequestData.irbFileHandleId]) {
+        requestedFileTypes[dataAccessRequestData.irbFileHandleId] = []
+      }
+      requestedFileTypes[dataAccessRequestData!.irbFileHandleId].push("IRB")
+      setIRB(requestObj)
+    }
+
+    if(dataAccessRequestData.attachments && dataAccessRequestData.attachments.length) {
+      dataAccessRequestData.attachments.forEach((id) => {
+        batchFileRequest.requestedFiles.push({
+          associateObjectId: dataAccessRequestData!.id,
+          associateObjectType: FileHandleAssociateType.DataAccessRequestAttachment,
+          fileHandleId: id,
+        })
+        if (!requestedFileTypes[id]) {
+          requestedFileTypes[id] = []
+        }
+        requestedFileTypes[id].push("attachments")
+      })
+    }
+
+    // Fetch the required doc file names and save them in the state variables
+    if (batchFileRequest.requestedFiles.length) {
+      getFiles(batchFileRequest, token).then(resp => {
+        resp.requestedFiles.forEach(file => {
+          const fileName = file.fileHandle!.fileName
+          const fileTypes = requestedFileTypes[file.fileHandleId]
+
+          fileTypes.forEach((type:string) => {
+            switch (type) {
+              case "DUCTemplate":
+                setDUCTemplate(prevState => {
+                  return Object.assign({}, prevState, {fileName: fileName})
+                })
+                requestedFileTypes[file.fileHandleId].splice(fileTypes.indexOf("DUCTemplate"), 1)
+                break
+              case "DUC":
+                setDUC(prevState => {
+                  return Object.assign({}, prevState, {fileName: fileName})
+                })
+                requestedFileTypes[file.fileHandleId].splice(fileTypes.indexOf("DUC"), 1)
+                break
+              case "IRB":
+                setIRB(prevState => {
+                  return Object.assign({}, prevState, {fileName: fileName})
+                })
+                requestedFileTypes[file.fileHandleId].splice(fileTypes.indexOf("IRB"), 1)
+                break
+              case "attachments":
+                setAttachments(prev => [...prev, {
+                  fileName: fileName,
+                  associateObjectId: dataAccessRequestData!.id,
+                  associateObjectType: FileHandleAssociateType.DataAccessRequestAttachment,
+                  fileHandleId: file.fileHandleId
+                }])
+                requestedFileTypes[file.fileHandleId].splice(fileTypes.indexOf("attachments"), 1)
+                break
+              default:
+            }
+          })
+        })
+      }) // end getFiles
+    }
+  }
+
+  const handleSubmit = async () => {
+    if (formSubmitRequestObject) {
+      try {
+        const resp = await updateDataAccessRequest(formSubmitRequestObject, token)
+        if (resp.etag) { // save success
```

Comment:
Same question: have you encountered a case when you have a response where etag is undefined?  I'd expect an error from the updateDataAccessRequest call.

Reply:
I added an else block to show an error notification if resp.etag is not defined.